x/nft/client/cli/query.go

```diff
@@ -0,0 +1,245 @@
+		Short:                      "Querying commands for the nft module",
+		Long:                       "",
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	nftQueryCmd.AddCommand(
+		GetCmdQueryClass(),
+		GetCmdQueryClasses(),
+		GetCmdQueryNFT(),
+		GetCmdQueryNFTs(),
+		GetCmdQueryOwner(),
+		GetCmdQueryBalance(),
+		GetCmdQuerySupply(),
+	)
+	return nftQueryCmd
+}
+
+// GetCmdQueryClass implements the query class command.
+func GetCmdQueryClass() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "class [class-id]",
+		Args:    cobra.ExactArgs(1),
+		Short:   "query an NFT class based on its id",
+		Example: fmt.Sprintf(`$ %s query %s class <class-id>`, version.AppName, nft.ModuleName),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+			queryClient := nft.NewQueryClient(clientCtx)
+			res, err := queryClient.Class(cmd.Context(), &nft.QueryClassRequest{
+				ClassId: args[0],
+			})
+			if err != nil {
+				return err
+			}
+			return clientCtx.PrintProto(res)
+		},
+	}
+	flags.AddQueryFlagsToCmd(cmd)
+	return cmd
+}
+
+// GetCmdQueryClasses implements the query classes command.
+func GetCmdQueryClasses() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "classes",
+		Short:   "query all NFT classes",
+		Example: fmt.Sprintf(`$ %s query %s classes`, version.AppName, nft.ModuleName),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+			queryClient := nft.NewQueryClient(clientCtx)
+			pageReq, err := client.ReadPageRequest(cmd.Flags())
+			if err != nil {
+				return err
+			}
+			res, err := queryClient.Classes(cmd.Context(), &nft.QueryClassesRequest{
+				Pagination: pageReq,
+			})
+			if err != nil {
+				return err
+			}
+			return clientCtx.PrintProto(res)
+		},
+	}
+	flags.AddQueryFlagsToCmd(cmd)
+	flags.AddPaginationFlagsToCmd(cmd, "classes")
+	return cmd
+}
+
+// GetCmdQueryNFT implements the query nft command.
+func GetCmdQueryNFT() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "nft [class-id] [nft-id]",
+		Args:    cobra.ExactArgs(2),
+		Short:   "query an NFT based on its class and id.",
+		Example: fmt.Sprintf(`$ %s query %s nft`, version.AppName, nft.ModuleName),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+			queryClient := nft.NewQueryClient(clientCtx)
+			res, err := queryClient.NFT(cmd.Context(), &nft.QueryNFTRequest{
+				ClassId: args[0],
+				Id:      args[1],
+			})
+			if err != nil {
+				return err
+			}
+			return clientCtx.PrintProto(res)
+		},
+	}
+	flags.AddQueryFlagsToCmd(cmd)
+	return cmd
+}
+
+// GetCmdQueryNFT implements the query nft command.
+func GetCmdQueryNFTs() *cobra.Command {
```

Comment:
How does one query just by owner? This seems a bit confusing and could be improved.

Reply:
The owner in _GetCmdQueryNFTs_ is an optional parameter. Now I change it to flag and pass it in. Will it be clearer than before?